src/components/molecules/ProjectCard.js

```diff
@@ -37,98 +39,97 @@ class ProjectCard extends Component {
-        
+        console.log(project_id, 'this is project_id');
         return axios.post(`/api/follow/${project_id}`)
             .then(res => {
+                console.log('Follow response', res);
                 this.props.updateProjects(project_id);
             });
     }
-    
-    setProjects() {
-        let iconColor = 'red';
-
-        if (this.props.projects.length > 0) {
-            const filters = (this.props.filters) ? (this.props.filters.filter(item => {
-                return item.length > 1;
-            })) : (null);
-            const projects = (Boolean(this.props.limit)) ? (
-                    this.props.projects
-                ) : (
-                    this.filterProjects(this.props.projects, filters)
-                );
-
-            return projects.map((item,i) => {
-                if (!this.props.limit || i < this.props.limit) {
+    render() {
+        const partial = Boolean(this.state.limit);
+        let projects = this.filterProjects(this.props.projects, this.state.filters);
 
-                    if (this.props.user) {
-                        // if user is loggedIn, check whether proect in followed
-                        iconColor = this.props.user.followedProjects.includes(item._id)
-                        ? "#B7140E"
-                        : "#9E9E9E"
-                    }
-                    
-                    return (
-                        <div className="col-md-3 card"
-                            onClick={() => this.props.history.push('/project/view/' + item._id)}
-                            key={i}
-                            id={item._id}>
-                            {this.props.user && this.props.user.projects.includes(item._id) === false && <FaStar 
-                                className={this.props.user.followedProjects.includes(item._id) ? "follow-icon active-icon" : "follow-icon"} 
-                                size={32} 
-                                color={iconColor} 
-                                onClick={this.handleClick.bind(this, item._id)}
-                            />}
-                            <div className="card-body">
-                                <p className="card-category">{item.category}</p>
-                                <h4 className="card-title">{item.title}</h4>
-                                    <Dotdotdot clamp={4}>
-                                    <p className="card-text">
-                                        {item.description}</p>
-                                    </Dotdotdot>
-                                <Button label="More" />
-                            </div>
-                        </div>  
-                    );      
-                } else {
-                    return null;
-                }
-            });
+        if (projects) {
+            return (
+                <div className={partial ? "container" : "container project-cards-full"}
+                >
+                    {(!partial) ? (
```

Comment:
This is just personal preference. But this could be written as `{!partial && <SearchBox />}` and have the same affect as using ternary.  Just saved you having to write an empty condition.